### ISSUE TITLE
Memory consumption

### ISSUE THREAD

**Issue by samanemami** [zzd1992/GBDTMO-EX#7]
I understand that you measured the model complexity in your paper, but, are there any related experiments for this matter (Memory usage)?

Also, you are using a dynamic lib for training, so how is it possible for your model to measure the memory in the `lib.so`?

Thanks in advance

**Comment by zzd1992** [comment]
I use `memory_profiler`.
See `https://pypi.org/project/memory-profiler/` for details.
The memory of dataset is not included.

**Comment by samanemami** [comment]
Thank you

Since the model is using a dynamic library in C++, is this [method ](https://pypi.org/project/memory-profiler/)also compute the library or only consider Python?

**Comment by zzd1992** [comment]
I believe that the memory of dynamic library is included. When we run model.train(), there is no memory introduced in Python. But the profiler shows memory becomes more by model.train().

I loss my code of this part. So I am not 100% sure about it.

**Comment by zzd1992** [comment]
when I compute the memory, the dataset is not included. I focus on the increment memory by the model.

**Comment by samanemami** [comment]
I also measured it with [tracemalloc()](https://docs.python.org/es/3/library/tracemalloc.html), by adding a tree in each epoch. I expected a linear growth, but the memory usage trend was so noisy.
I also, considered only the train method.

**Comment by samanemami** [comment]
We have the same situation for the time as well. As the trend line is not linear and there is something inside the library that the time() could not process. By trend line I mean something as follows;
```Python
for i in range(1, n):
	model = GBDTMulti(LIB, out_dim='out_dim', params=**params)
        t0 = time()
	model.train(i)
        t1 = time() - t0
```

**Comment by samanemami** [comment]
I believe that the model uses multiprocessing which duplicates the process memory, so if the it uses histogram strategy of lightGBM, it should consume more memory, at least more than the XGboost.
